tools/serverpod_cli/lib/src/language_server/language_server.dart

```diff
@@ -39,36 +111,56 @@ Future<void> runLanguageServer() async {
       );
     });
 
-    var diagnostics = _validateYamlProtocol(
-      contentChanges.last.text,
-      params.textDocument.uri.toString(),
+    statefulAnalyzer.validateProtocol(
+      contentChanges.first.text,
+      params.textDocument.uri,
     );
 
-    connection.sendDiagnostics(
-      PublishDiagnosticsParams(
-        diagnostics: diagnostics,
-        uri: params.textDocument.uri,
-      ),
-    );
+    // This can be optimised to only validate the files we know have related errors.
+    statefulAnalyzer.validateAll();
   });
 
   await connection.listen();
 }
 
-List<Diagnostic> _validateYamlProtocol(String yaml, String sourcePath) {
-  var collector = CodeGenerationCollector();
+Future<GeneratorConfig> _initializeAnalyzer(
```

Comment:
Suggestion: Either separate the concerns of initialising the analysera and generating the config or provide a better name for the method.

This method does two things. Looks like we could create the config without having the analyzer available.

Reply:
Yes agreed, but postponing.